server/routerlicious/packages/services-utils/src/auth.ts

```diff
@@ -291,6 +292,8 @@ export function verifyStorageToken(
 				tenantManager,
 				moreOptions,
 			);
+			// Riddler is known to take too long sometimes. Check timeout before continuing.
+			getGlobalTimeoutContext().checkTimeout();
```

Comment:
What is the expected behavior in the following scenario: say that `verifyToken()` takes 2 minutes to complete, and our timeout is 30 seconds. By the time we get to line 296 and call `getGlobalTimeoutContext().checkTimeout();`, the timeout would already have happened. In that case, we would throw an appropriate error code (503), but that would happen after the client potentially already disconnected from the service

Reply:
Yup! That's correct. So this would throw, causing us to stop processing the request, and we would internally log a 503 error, but it wouldn't reach the client.